azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/storage/StorageAutoConfiguration.java

```diff
@@ -39,26 +42,42 @@ public StorageAutoConfiguration(StorageProperties properties) {
     }
 
     /**
-     * Declare CloudStorageAccount bean.
-     *
-     * @return CloudStorageAccount bean
+     * @param options PipelineOptions bean, not required.
+     * @return
      */
     @Bean
-    @Scope("prototype")
-    public CloudStorageAccount cloudStorageAccount() throws URISyntaxException, InvalidKeyException {
-        LOG.debug("cloudStorageAccount called");
+    public ServiceURL createServiceUrl(@Autowired(required = false) PipelineOptions options) throws InvalidKeyException,
+            MalformedURLException {
+        LOG.debug("Creating ServiceURL bean...");
         trackCustomEvent();
-        return createCloudStorageAccount();
+
+        final SharedKeyCredentials credentials = new SharedKeyCredentials(properties.getAccountName(),
+                properties.getAccountKey());
+        final URL blobUrl = new URL(String.format(BLOB_URL, properties.getAccountName()));
+        final PipelineOptions pipelineOptions = buildOptions(options);
+        final ServiceURL serviceURL = new ServiceURL(blobUrl, StorageURL.createPipeline(credentials, pipelineOptions));
+
+        return serviceURL;
     }
 
-    /**
-     * Helper function for creating CloudStorageAccount instance from storage connection string.
-     *
-     * @return CloudStorageAccount object
-     */
-    private CloudStorageAccount createCloudStorageAccount() throws URISyntaxException, InvalidKeyException {
-        LOG.debug("createCloudStorageAccount called");
-        return CloudStorageAccount.parse(properties.getConnectionString());
+    private PipelineOptions buildOptions(PipelineOptions fromOptions) {
+        final PipelineOptions pipelineOptions;
+        if (fromOptions != null) {
+            pipelineOptions = fromOptions;
+        } else {
+            pipelineOptions = new PipelineOptions();
+        }
```

Comment:
As the if-else is simple, suggest to convert to `option = fromOption != null ? fromOPtion : new option()`

Reply:
Updated expression.